### PR TITLE
Document the robj_ndarray module

### PR DESCRIPTION
**Pull request by multimeric**
* Document the robj_ndarray module 
* Add conversion for owned arrays, with tests

### CODE REVIEW

**Comment by multimeric** [comment]
Two questions for whoever reviews this:

1. Why is the WIndows build failing? Is that at all relating to my PR?
2. Why does it seem like the doctests aren't being run for this PR. I've added a number of examples that do run correctly and it would be nice to know that they're being checked.

File: extendr-api/src/robj_ndarray.rs
Comment (Ilia-Kosenkov) [inline extendr-api/src/robj_ndarray.rs:10]: In #443 I am trying to add support for `TryFrom<Robj>`, which is the proper way to handle conversions. Should we document it instead of obsolete `FromRobj` or wait till I merge my PR?

Comment (multimeric) [reply extendr-api/src/robj_ndarray.rs:10]: For the sake of simplicity, maybe we just leave my PR as is, and then we can update the link in your PR? Unless yours gets merged first, in which case I'll do the fix here.

**Review by Ilia-Kosenkov — APPROVED** [review]
Ok, LGTM

**Comment by constantingoeldel** [comment]
Thanks for the documentation, this already helped me!

**Comment by multimeric** [comment]
Going to hold off on merging this until we merge #460, so I can rewrite to use `rstest`.

**Comment by multimeric** [comment]
Has now been updated to use `rstest`. Once the CI is running again, this is good to go.

**Review by Ilia-Kosenkov — CHANGES_REQUESTED** [review]
How about we also mention in the docs that we can handle `ArrayView<Rint>` and friends?

**Review by Ilia-Kosenkov — APPROVED** [review]

**Pull request merged**
